nucypher/characters/chaotic.py

```diff
@@ -0,0 +1,290 @@
+
+            checksum_addresses = [gen_eth_addr(i) for i in range(0, self.shares_num)]
+
+            self.fake_ritual = _Blasphemy(
+                tau=self.tau,
+                threshold=self.security_threshold,
+                shares_num=self.shares_num,
+                checksum_addresses=checksum_addresses,
+            )
+
+            validator_keypairs = [
+                ferveo.Keypair.random() for _ in range(self.shares_num)
+            ]
+
+            validators = [
+                ferveo.Validator(checksum_addresses[i], keypair.public_key())
+                for i, keypair in enumerate(validator_keypairs)
+            ]
+
+            # Validators must be sorted by their public key
+            validators.sort(key=attrgetter("address"))
+
+            # Each validator holds their own DKG instance and generates a transcript every
+            # validator, including themselves
+            self.aggregation_messages = []
+            for sender in validators:
+                dkg = ferveo.Dkg(
+                    tau=self.tau,
+                    shares_num=self.shares_num,
+                    security_threshold=self.security_threshold,
+                    validators=validators,
+                    me=sender,
+                )
+                self.aggregation_messages.append(
+                    ferveo.ValidatorMessage(sender, dkg.generate_transcript())
+                )
+
+            self.dkg = dkg
+            self.validators = validators
+            self.validator_keypairs = validator_keypairs
+
+            self.server_aggregate = dkg.aggregate_transcripts(self.aggregation_messages)
+            assert self.server_aggregate.verify(
+                self.shares_num, self.aggregation_messages
+            )
+
+            # And the client can also aggregate and verify the transcripts
+            self.client_aggregate = ferveo.AggregatedTranscript(
+                self.aggregation_messages
+            )
+            assert self.server_aggregate.verify(
+                self.shares_num, self.aggregation_messages
+            )
+
+    _dkg_insight = DKGInsight()
+
+
+class NiceGuyEddie(Enrico, DKGOmniscient):
+    """
+    Like Enrico, but from Reservoir Dogs.
+
+    He doesn't know who's shot, who's not.
+    """
+
+    def __init__(self, encrypting_key, *args, **kwargs):
+        del encrypting_key  # We take this to match the Enrico public API, but we don't use it, because...
+
+        # ...we're going to use the DKG public key as the encrypting key, and ignore the key passed in.
+        encrypting_key_we_actually_want_to_use = self._dkg_insight.dkg.public_key
+        super().__init__(
+            # https://imgflip.com/i/7o0po4
+            encrypting_key=encrypting_key_we_actually_want_to_use,
+            *args,
+            **kwargs,
+        )
+
+
+class DKGOmniscientDecryptionClient(ThresholdDecryptionClient):
+    def gather_encrypted_decryption_shares(
+        self,
+        encrypted_requests,
```

Comment:
typehint would be nice here:

```suggestion
        encrypted_requests: Dict[ChecksumAddress, EncryptedThresholdDecryptionRequest],
```